charts/kong/values.yaml

```diff
@@ -146,7 +146,7 @@ admin:
   # Enable creating a Kubernetes service for the admin API
   # Disabling this is recommended for most ingress controller configurations
   # Enterprise users that wish to use Kong Manager with the controller should enable this
-  enabled: false
+  enabled: true
   type: NodePort
```

Comment:
Given that Service Discovery is now to become a default, I think we should use `ClusterIP` type with `spec.clusterIP` set to `None`. `NodePort` would unnecessarily allocate ports while they're not needed from the KIC perspective. Using `NodePort` would also make the attack surface unnecessarily bigger by default.

>  headless Services are a annoying to check for diagnostic purposes.

Could you elaborate on this more? As I understand there should not be a problem to have a port-forward set up for a headless Service. 🤔

Reply:
I generally use LoadBalancer Services with metallb because they're easier to use than port-forward (you can always send to the same thing, regardless of whether you've restarted the Pod that was handling your port-forward). `None` ClusterIPs can't be converted to an assigned ClusterIP AFAICT, so you can't easily convert them to something else unless you started with the other configuration.

I also tend to be lazy, install with defaults, and edit my resources in place rather than setting them up in advance. There's no particular reason for this other than preference (I like working with the complete resource YAML instead of building a values.yaml), but unless we need `None` I'd prefer to avoid it.

Reply:
Yeah, I think we can stay with `ClusterIP` without `None` - that's not required. KIC uses its `EndpointSlices` anyway which are populated in both cases.